tests/model_fields/tests.py

```diff
@@ -661,6 +689,15 @@ def test_backend_range_validation(self):
             }
             with self.assertRaisesMessage(ValidationError, expected_message):
                 instance.full_clean()
+            try:
+                with transaction.atomic():
+                    instance.save()
+            except (DataError, IntegrityError):
+                pass  # backends such as Postgres don't fail silently
+            else:
+                qs = self.model.objects.filter(value__lte=min_value)
```

Comment:
So pseudo code wise should it look like

```python
if truncates:
    with transaction.atomic():
        instance.save()
    qs = self.model.objects.filter(value__lte=min_value)
    self.assertEqual(qs.count(), 1)
    self.assertEqual(qs[0].value, min_value)
else:
    with transaction.atomic(), assertRaises():
        instance.save()
```

Can you give a better idea of what you had in mind for `truncates_out_of_range_x`? As far as word choice truncate is really concise to be inclusive of the word range.